### PR TITLE
Fix pytest configuration and CXXFLAGS/CPPFLAGS

### PR DESCRIPTION
**Pull request by bjlittle**
This PR backs out commit `a0d3fa626d5ad6bf3838f2dca24d6fc272af90e2`.

It replaces it with the configuration required for `pytest` to discover the root of the `mint` package.

Notably, this PR also addresses the building issues experienced when attempting to build `mint` on `conda-forge`, see https://github.com/conda-forge/staged-recipes/pull/14354

This simply involves using the `CXXFLAGS` over the `CPPFLAGS` when cythonizing in the `setup.py`.

I've also rationalised the `setup.py` by moving most of the content into the `setup.cfg`, as there is motivation to do this from [setuptools](https://setuptools.readthedocs.io/en/latest/userguide/declarative_config.html).

This PR also incorporates CI testing on `cirrus-ci`, specifically for

- `py36`, `py37` and  `py38` on `linux-64`, and
- `py38` on `osx-64`

I don't have `admin` rights to the repo so you're going to have to register the repo with `cirrus-ci`. It's pretty simple to do. For further details see [here](https://cirrus-ci.org/guide/quick-start/).

Also, see [here](https://cirrus-ci.com/build/5073525984722944) for the `cirrus-ci` run for this PR on my branch.

Later I can add a badge on the `README.md` for convenience to show the build state for `master` and also take you to the `cirrus-ci` dashboard.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ conda activate mint-dev
 
 In the root MINT directory then type:
 ```
-python setup.py install 
+pip install --no-deps --editable .
```
Comment (bjlittle) [inline README.md:43]: @pletzer `mint` should be installed in `develop` mode

File: requirements/mint.yml [hunk 2]
```diff
@@ -4,14 +4,15 @@ channels:
   - conda-forge
 
 dependencies:
-  - python=3.9
+  - python=3.8  # vtk for py39 is not available from PyPI @March 2021
```
Comment (bjlittle) [inline requirements/mint.yml:7]: @pletzer We're limited to `py38` for now, as we're building `mint` using PEP517/518, we therefore require to use the `vtk` package from PyPI and that's limited to `py38` (latest) at the moment

File: setup.cfg [hunk 3]
```diff
@@ -1,3 +1,59 @@
+[tool:pytest]
+testpaths =
+    mint/
+addopts =
+    -ra
+    -v
```
Comment (bjlittle) [inline setup.cfg:6]: @pletzer This configures `pytest` to run the `develop` installed version of `mint`

File: setup.py [hunk 4]
```diff
@@ -76,20 +74,22 @@ def getCondaNetCDF():
...
 cpp_flags = os.getenv("CPPFLAGS")
-if cpp_flags:
-    cpp11_flag = cpp_flags  # on Windows: '/std:c11'
+cxx_flags = os.getenv("CXXFLAGS")
+if cxx_flags is not None:
+    extra_compile_args = cxx_flags.split()
+elif cpp_flags is not None:
+    extra_compile_args = cpp_flags.split()
+else:
+    extra_compile_args = ["-std=c++11"]
```
Comment (bjlittle) [inline setup.py:92]: Preference use of `CXXFLAGS` over `CPPFLAGS` over default of `-std=c++11`.

As we're building c++, this make sense given the standards.

**Comment by pletzer** [comment]
Thanks @bjlittle ! Works beautifully

**Pull request merged**